routes/items.js

```diff
@@ -1,31 +1,58 @@
 const { Router } = require("express");
 const router = Router();
-
 const itemData = require('../dataInterface/items');
 
+// This NPM Package aid in parsing the request body
+const bodyParser = require('body-parser');
+// router?app.use(something()) tells the server to use this function throug it
+router.use(bodyParser())
+
 router.get("/", (req, res, next) => {
   res.status(200).send(itemData.getAll())
 });
 
 router.get("/:id", (req, res, next) => {
-  // TODO: complete writing this route handler
-  res.status(501).send({ error: 'route not yet implemented' });
+  const queryExist = itemData.getById(req.params.id)
+  if(queryExist){
+    res.status(200).json(queryExist)
+  }else{
+    res.status(404).send({error: "not found"})
+  }
 });
 
 router.post("/", (req, res, next) => {
-  itemData.create(req.body);
-  res.sendStatus(200);
+  if(req.body.field != ""){
+    itemData.create(req.body);
+    res.sendStatus(200);
+  }else{
+    res.sendStatus(400)
+  }
+
 });
 
 router.put("/:id", (req, res, next) => {
-  // TODO: complete writing this route handler
-  res.status(501).send({ error: 'route not yet implemented' });
+  const queryExist = itemData.getById(req.params.id)
```

Comment:
This logic of checking if a sent id is associated with a real object should be the responsibility of the data interface method. We want to keep route handlers as ignorant as possible of data related business logic.